releasenotes/notes/rename-user_parameters-to-training_parameters-7d8cf1e056b967a7.yaml

```diff
@@ -0,0 +1,15 @@
+---
+issues:
+  - |
+    Rename ``user_parameters`` in class ``QuantumKernel`` to ``training_parameters``.
+    This means rename the keyword-argument in the constructor, the properties
+    ``user_parameters``, ``user_param_binds``, the methods ``assign_user_parameters``
+    ``bind_user_parameters``, ``get_unbound_user_parameters``, to
+    ``training_parameters``, ``trainin_param_binds``, ``assign_training_parameters``
+    ``bind_training_parameters``, ``get_unbound_training_parameters``.
+
+deprecations:
+  - |
+    The existing keyword-argument ``user_parameters`` and the properties
+    ``user_parameters``, ``user_param_binds``, the methods ``assign_user_parameters``
+    ``bind_user_parameters``, ``get_unbound_user_parameters`` are all deprecated.
```

Comment:
The `issues` section is to cover known bugs/problems that are in the release. I think this note can be just done as a deprecations section. And I am thinking something like this which more or less combines the above as well as provides a bit of explanation.

```
    For the class :class:`~qiskit_machine_learning.kernels.QuantumKernel`, to
    improve usability and better describe the usage, ``user_parameters`` has been
    renamed to ``training_parameters``; current behavior is retained. For this
    change the constructor parameter ``user_parameters`` is now deprecated and
    replaced by ``training_parameters``. The related properties and methods are
    renamed to match. Thus the properties ``user_parameters`` and ``user_param_binds`
    are both deprecated and replaced by ``training_parameters`` and
    ``training_param_binds`` respectively. The methods ``assign_user_parameters``,
    ``bind_user_parameters`` and ``get_unbound_user_parameters`` are deprecated and
    replaced by ``assign_training_parameters``, ``bind_training_parameters`` and
    ``get_unbound_training_parameters``.
```
Its possible to link the methods  but I figured linking to the class would suffice. I also corrected a typo in training i.e. `trainin_param_binds` 

While changing the naming do we want to consider `training_param_binds? We have parameters spelled out in the others and some are longer e.g `get_unbound_user_parameters` would not this be better as say `training_parameters_binding` or something more consistent like that? I note this is a property versus get_unbound_user_parameters as a method. The latter could be made a property as `unbound_user_parameters` as its hard for me to see really why one is a property and the other method.  @calebj15 for any thoughts here

Reply:
> While changing the naming do we want to consider `training_param_binds? We have parameters spelled out in the others and some are longer e.g `get_unbound_user_parameters`would not this be better as say`training_parameters_binding`or something more consistent like that? I note this is a property versus get_unbound_user_parameters as a method. The latter could be made a property as`unbound_user_parameters` as its hard for me to see really why one is a property and the other method. @calebj15 for any thoughts here

unbound_user_parameters isn't really a field that is kept up with, so I didn't see it as a property. It is something that is calculated on the fly and returned, so I felt that a class method was more appropriate there

Reply:
> so I felt that a class method was more appropriate there

Sure thats fine. For me its more how the end-user engages with the class rather than if there is computation that goes on behind the scenes for the result. If its significant computation then a method though is probably the better choice as accessing a property should feel no so different than accessing a public instance variable for me.

Reply:
Re: release-note, I added a bullet-list.  I agree that it's much better than the text; shall I remove most of the textual list in favor of the bullet-list?

Reply:
OK, I cleaned-up the release-note (removed most of the text), the bullet-list is much nicer.

Reply:
Any comment on this aspect from above? Its the only name that has `parameters` abbreviated to `param`

> While changing the naming do we want to consider training_param_binds? We have parameters spelled out in the others and some are longer e.g get_unbound_user_parameters Would not this be better as say training_parameters_binding or something more consistent like that?

Reply:
The parameter is already being renamed under this PR from `user_param_binds` to `training_param_binds`. All this would do is to change the new name a little more, thats all I think!

Reply:
Yes, I agree. Let's go with training_parameter_binds!

Reply:
Roger that, will make the change.